File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-cd-5-4/PCDeliverableObligationCharac.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class PCDeliverableObligationCharac : ISerialized
    {
        public PCDeliverableObligationCharac(XmlNode xmlNode)
        {
            XmlNode applicableNode = xmlNode.SelectSingleNode("applicable");
            
            if (applicableNode != null)
            {
                if (applicableNode.Attributes["href"] != null || applicableNode.Attributes["id"] != null) 
                {
                    if (applicableNode.Attributes["id"] != null) 
                    {
                        applicableIDRef_ = applicableNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(applicableNode);
                        IDManager.SetID(applicableIDRef_, ob);
                    }
                    else if (applicableNode.Attributes["href"] != null)
                    {
                        applicableIDRef_ = applicableNode.Attributes["href"].Value;
                    }
                    else
                    {
                        applicable_ = new XsdTypeBoolean(applicableNode);
                    }
                }
                else
                {
                    applicable_ = new XsdTypeBoolean(applicableNode);
                }
            }
            
        
            XmlNode partialCashSettlementNode = xmlNode.SelectSingleNode("partialCashSettlement");
            
            if (partialCashSettlementNode != null)
            {
                if (partialCashSettlementNode.Attributes["href"] != null || partialCashSettlementNode.Attributes["id"] != null) 
                {
                    if (partialCashSettlementNode.Attributes["id"] != null) 
                    {
                        partialCashSettlementIDRef_ = partialCashSettlementNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(partialCashSettlementNode);
                        IDManager.SetID(partialCashSettlementIDRef_, ob);
                    }
                    else if (partialCashSettlementNode.Attributes["href"] != null)
                    {
                        partialCashSettlementIDRef_ = partialCashSettlementNode.Attributes["href"].Value;
                    }
                    else
                    {
                        partialCashSettlement_ = new XsdTypeBoolean(partialCashSettlementNode);
                    }
                }
                else
                {
                    partialCashSettlement_ = new XsdTypeBoolean(partialCashSettlementNode);
                }
            }
            
        
        }
        
    
        #region Applicable_
        private XsdTypeBoolean applicable_;
        public XsdTypeBoolean Applicable_
        {
            get
            {
                if (this.applicable_ != null)
                {
                    return this.applicable_; 
                }
                else if (this.applicableIDRef_ != null)
                {
                    applicable_ = IDManager.getID(applicableIDRef_) as XsdTypeBoolean;
                    return this.applicable_; 
                }
                else
                {
                      return this.applicable_; 
                }
            }
            set
            {
                if (this.applicable_ != value)
                {
                    this.applicable_ = value;
                }
            }
        }
        #endregion
        
        public string applicableIDRef_ { get; set; }
        #region PartialCashSettlement_
        private XsdTypeBoolean partialCashSettlement_;
        public XsdTypeBoolean PartialCashSettlement_
        {
            get
            {
                if (this.partialCashSettlement_ != null)
                {
                    return this.partialCashSettlement_; 
                }
                else if (this.partialCashSettlementIDRef_ != null)
                {
                    partialCashSettlement_ = IDManager.getID(partialCashSettlementIDRef_) as XsdTypeBoolean;
                    return this.partialCashSettlement_; 
                }
                else
                {
                      return this.partialCashSettlement_; 
                }
            }
            set
            {
                if (this.partialCashSettlement_ != value)
                {
                    this.partialCashSettlement_ = value;
                }
            }
        }
        #endregion
        
        public string partialCashSettlementIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-cd-5-4/PCDeliverableObligationCharac.cpp
// PCDeliverableObligationCharac.cpp 
#include "PCDeliverableObligationCharac.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

PCDeliverableObligationCharac::PCDeliverableObligationCharac(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //applicableNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* applicableNode = xmlNode->FirstChildElement("applicable");

   if(applicableNode){applicableIsNull_ = false;}
   else{applicableIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- applicableNode , address : " << applicableNode << std::endl;
   #endif
   if(applicableNode)
   {
      if(applicableNode->Attribute("href") || applicableNode->Attribute("id"))
      {
          if(applicableNode->Attribute("id"))
          {
             applicableIDRef_ = applicableNode->Attribute("id");
             applicable_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(applicableNode));
             applicable_->setID(applicableIDRef_);
             IDManager::instance().setID(applicableIDRef_,applicable_);
          }
          else if(applicableNode->Attribute("href")) { applicableIDRef_ = applicableNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { applicable_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(applicableNode));}
   }

   //partialCashSettlementNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* partialCashSettlementNode = xmlNode->FirstChildElement("partialCashSettlement");

   if(partialCashSettlementNode){partialCashSettlementIsNull_ = false;}
   else{partialCashSettlementIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partialCashSettlementNode , address : " << partialCashSettlementNode << std::endl;
   #endif
   if(partialCashSettlementNode)
   {
      if(partialCashSettlementNode->Attribute("href") || partialCashSettlementNode->Attribute("id"))
      {
          if(partialCashSettlementNode->Attribute("id"))
          {
             partialCashSettlementIDRef_ = partialCashSettlementNode->Attribute("id");
             partialCashSettlement_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(partialCashSettlementNode));
             partialCashSettlement_->setID(partialCashSettlementIDRef_);
             IDManager::instance().setID(partialCashSettlementIDRef_,partialCashSettlement_);
          }
          else if(partialCashSettlementNode->Attribute("href")) { partialCashSettlementIDRef_ = partialCashSettlementNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { partialCashSettlement_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(partialCashSettlementNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeBoolean> PCDeliverableObligationCharac::getApplicable()
{
   if(!this->applicableIsNull_){
        if(applicableIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(applicableIDRef_));
        }else{
             return this->applicable_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> PCDeliverableObligationCharac::getPartialCashSettlement()
{
   if(!this->partialCashSettlementIsNull_){
        if(partialCashSettlementIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(partialCashSettlementIDRef_));
        }else{
             return this->partialCashSettlement_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
}

